src/test/java/scm/address/logic/parser/AddScheduleCommandParserTest.java

```diff
@@ -0,0 +1,159 @@
+    }
+
+    @Test
+    public void parse_validArgs_returnsAddScheduleCommand() {
+        LocalDateTime startDateTime = LocalDateTime.now();
+        LocalDateTime endDateTime = startDateTime.plusHours(1);
+        String formattedStart = startDateTime.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+        String formattedEnd = endDateTime.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+
+        String input = String.format("title/Meeting d/Discuss project start/%s end/%s",
+                formattedStart, formattedEnd);
+
+        AddScheduleCommand expectedCommand = new AddScheduleCommand(
+                new Schedule(new Title("Meeting"), new Description("Discuss project"),
+                        startDateTime.format(formatter), endDateTime.format(formatter))
+        );
+
+        AddScheduleCommandParser parser = new AddScheduleCommandParser();
+        try {
+            AddScheduleCommand actualCommand = parser.parse(input);
+            assertEquals(expectedCommand, actualCommand);
+        } catch (ParseException e) {
+            // No exception expected
+        }
+    }
+
+    @Test
+    public void isDateTimeBefore_otherDateTime_returnsTrue() {
+        String datetimeStr1 = "2023-03-21T14:00:00";
+        String datetimeStr2 = "2023-03-21T15:00:00";
+
+        LocalDateTime dateTime1 = LocalDateTime.parse(datetimeStr1);
+        LocalDateTime dateTime2 = LocalDateTime.parse(datetimeStr2);
+
+        assertTrue(dateTime1.isBefore(dateTime2));
+    }
+
+    @Test
+    public void areAllPrefixesPresent_allPresent_returnsTrue() {
+        ArgumentMultimap argumentMultimap = new ArgumentMultimap();
+        argumentMultimap.put(new Prefix("title/"), "Meeting");
+        argumentMultimap.put(new Prefix("d/"), "Discuss project");
+        argumentMultimap.put(new Prefix("start/"), "2023-03-21 15:00");
+        argumentMultimap.put(new Prefix("end/"), "2023-03-21 16:00");
+
+        Prefix titlePrefix = new Prefix("title/");
+        Prefix descriptionPrefix = new Prefix("d/");
+        Prefix startPrefix = new Prefix("start/");
+        Prefix endPrefix = new Prefix("end/");
+
+        boolean allPrefixesPresent = arePrefixesPresent(argumentMultimap,
+                titlePrefix, descriptionPrefix, startPrefix, endPrefix);
+
+        assertTrue(allPrefixesPresent);
+    }
+
+    @Test
+    public void parse_missingPrefixes_throwsParseException() {
+        String input = "title/Meeting d/Project discussion start/2023-03-21 15:00";
+        AddScheduleCommandParser parser = new AddScheduleCommandParser();
+
+        assertThrows(ParseException.class, () -> parser.parse(input));
+    }
+
+    @Test
+    public void isFirstDateTimeBeforeSecond_validDateTimes_correctResult() {
+        DateTimeComparison comparison = new DateTimeComparison();
+
+        assertFalse(comparison.isFirstDateTimeBeforeSecond("2023-03-21 15:00", "2023-03-21 16:00"));
```

Comment:
Constants can also be used here, I think?

Reply:
By constants do you mean variables that have been hardcoded? If it is then I think this one is ok

Reply:
Sure, no worries!

Reply:
Nevermind I realised what you are talking about now